Sources/PluginLibrary/Descriptor.swift

```diff
@@ -98,46 +134,66 @@ public final class FileDescriptor {
     self.services.forEach { $0.bind(file: self, registry: registry) }
   }
 
+  // TODO(thomasvl): Eventually hide this and just expose it info off the descriptors so
+  // paths aren't needed externally.
+  public func sourceCodeInfoLocation(path: [Int32]) -> Google_Protobuf_SourceCodeInfo.Location? {
+    guard let location = locationMap[HashableArray(path)] else {
```

Comment:
Could we reuse `Foundation.IndexPath` here? (We'd need to clean up the `Int32` to `Int` conversions.)